### ISSUE TITLE
lsp-java-treemacs.el dependencies missing

### ISSUE THREAD

**Issue by peterwvj** [emacs-lsp/lsp-java#101]
The `Package-Requires` header is missing from lsp-java-treemacs.el. In particular, the treemacs dependencies are not listed. I suspect this is why I'm getting errors like the one below:

```
Compiling file /home/peter/.emacs.d/elpa/lsp-java-20190208.553/lsp-java-treemacs.el at Mon Feb 11 08:18:19 2019
Entering directory ‘/home/peter/.emacs.d/elpa/lsp-java-20190208.553/’
lsp-java-treemacs.el:32:1:Error: Cannot open load file: No such file or directory, treemacs
```

**Comment by yveszoundi** [comment]
@peterwvj, I believe that the dependency is meant to be "optional" and I don't think that `lsp-java-treemacs` is provided as a dedicated package. 

Those errors are harmless, especially if you don't care about `treemacs`, I personally don't use it. Not sure if it's work adding a gigantic condition to guard against code configuring `treemacs` support (check first if `treemacs` is installed on the user machine).

**Comment by yyoncho** [comment]
@peterwvj, @yveszoundi is right - it was meant to be an optional dependency and you may ignore that error or install treemacs if you want to remove it. I am closing it as won't fix, please reopen or provide a PR if you disagree.

**Comment by peterwvj** [comment]
I agree that the lsp-java-treemacs.el/treemacs should be optional (I don't use it). However, installing lsp-java via Melpa forces me to download lsp-java-treemacs as well. I think the reason for this is found in the Melpa recipe, specifically that `:files` uses `:default`which forces me to download `lsp-java-treemacs.el` which I don't need.

```
(lsp-java :repo "emacs-lsp/lsp-java" :fetcher github :files (:defaults "icons"))
```
I understand that the compilation error is harmless, but it's still a clear indication that something is wrong. I mean a user who is installing `lsp-java` but who doesn't want to use treemacs should never see a compilation error like this. I don't think that simply saying that the error is harmless is acceptable.

I think you can fix this by submitting `lsp-java-treemacs` as a separate package to melpa. Only the maintainers can do that. Once that is done the `Package-Requires` header should be added in lsp-java-treemacs.el (it really is a mistake leaving this dependency out).

**Comment by yyoncho** [comment]
>  I don't think that simply saying that the error is harmless is acceptable.

I am trying to prioritize the work on lsp-java/dap-mode/lsp-mode and there a lot of stuff to be done. As you can see there are 30+ server methods that are not exposed only in lsp-java.

> Only the maintainers can do that.

Anyone could do that and CC the maintainers. I am reopening the bug and please let me know if you want to work on melpa onboarding+updating the docs or making the treemacs dependency optional to pass the byte compilation.

**Comment by yyoncho** [comment]
lsp-java-treemacs is no longer present - this closing.
